### PR TITLE
Controller configmap refactor

### PR DESCRIPTION
**Pull request by JoshVanL**
This PR refactors the controller to reduce the memory usage by only caching ConfigMap metadata, and not attempt to update/create ConfigMaps in terminating Namespaces.

Adds a Subscription channel so that the controller will reconcile on ConfigMaps in all Namespaces when the root CA changes.

/assign @irbekrm 

```release-note
NONE
```

### CODE REVIEW

File: pkg/tls/tls.go [hunk 1]
```diff
@@ -276,6 +299,12 @@ func (p *Provider) fetchCertificate(ctx context.Context) (time.Time, error) {
 	// If we are not using a custom root CA, then overwrite the existing with
 	// what was responded.
 	if len(p.opts.RootCACertFile) == 0 {
+		// If the root CAs bundle has been changed, send event to subscribers.
+		if !bytes.Equal(p.rootCA, bundle.CA) {
```
Comment (irbekrm) [inline pkg/tls/tls.go:303]: Question: do we not want to send these event _after_ lines 315-322 where we could potentially error out of this func if the root CA was not decodeable?

**Review by irbekrm — CHANGES_REQUESTED** [review]
This looks great to me- the usage of a Watch with `Source` is nice 👍🏼 

I've added a couple nits and suggestions, else happy to approve.

File: pkg/controller/configmap.go [hunk 2]
```diff
@@ -0,0 +1,209 @@
...
+	// Only reconcile config maps that match the well known name
+	if err := ctrl.NewControllerManagedBy(mgr).
+		// Reconcile ConfigMaps but only cache metadata
+		For(new(corev1.ConfigMap), builder.OnlyMetadata, builder.WithPredicates(predicate.NewPredicateFuncs(func(obj client.Object) bool {
+			// Only process ConfigMaps with the istio configmap name
+			return obj.GetName() == configMapNameIstioRoot
+		}))).
+
+		// Watch all Namespaces. Cache whole Namespace to include Phase Status.
+		Watches(&source.Kind{Type: new(corev1.Namespace)}, handler.EnqueueRequestsFromMapFunc(
```
[the remark this answers is not in the document]
Comment (JoshVanL) [reply pkg/controller/configmap.go:94]: It's fairly strongly worded by the controller-runtime docs to _not_ do something like this

https://github.com/kubernetes-sigs/controller-runtime/blob/master/FAQ.md#q-how-do-i-have-different-logic-in-my-reconciler-for-different-types-of-events-eg-create-update-delete

[the remark this answers is not in the document]
Comment (irbekrm) [reply pkg/controller/configmap.go:94]: Cool, didn't know this 👍🏼

**Review by irbekrm — APPROVED** [review]
/lgtm

**Pull request merged**